### ISSUE TITLE
Client examples?

### ISSUE THREAD

**Issue by dustinsmith1024** [go-swagger/go-swagger#76]
Is there an example of how to use the generated client? Struggling a bit on what to do.

**Comment by casualjim** [comment]
you mean like after generating, how to use it to make calls to the API you generated a client for?

I have as top priority now writing a documentation site, I will do this this weekend.
I'll try to write a sample for it in a couple of hours, once the day job ends :)

**Comment by dustinsmith1024** [comment]
Yea, I just an example of how to use it.  Kind of confused on what the `Transport` should be in the generated `New` method.  

Thanks @casualjim.

**Comment by casualjim** [comment]
you want to pass in this guy:

https://github.com/go-swagger/go-swagger/blob/master/httpkit/client/runtime.go#L31

The idea is that there are other transports later on like nanomsg, zeromq, websocket and so on.

**Comment by dustinsmith1024** [comment]
Thanks I got that working now. However getting authentication errors.

I followed this example https://github.com/go-swagger/go-swagger/blob/2e74a8626dbd66f60ef5a47e1d0bfaebd2c2b080/fixtures/codegen/tasklist.basic.yml and have this in my config.

```yaml 
securityDefinitions:
  auth_token:
    type: apiKey
    name: Authorization
    in: header
```

in the individual route i have:
```yaml
paths:
  /customers:
    get:
      summary: Customer List
      operationId: allCustomers
      description: All registered customers. 
      parameters:
      - name: order
        in: query
        type: string
        description: Any field in the database.
      tags:
        - Customers
      security:
        - auth_token: []
      responses:
        '200':
          description: An array of customers
          schema:
            type: array
            items:
              $ref: '#/definitions/CustomersResponse'
        default:
          description: Unexpected error
          schema:
            $ref: '#/definitions/Error'
```

This doesn't seem to generate any Authentication methods for me?  Or just not sure where to find them. I found this issue https://github.com/go-swagger/go-swagger/issues/40 but I dont have a `AuthenticatesFor` method in my generated code.

Thanks for the help.

**Comment by casualjim** [comment]
example is here: http://go-swagger.github.io/generate/client/

**Comment by dlazerka** [comment]
@casualjim Both links are broken now (404). Any new links please?

**Comment by casualjim** [comment]
https://goswagger.io/generate/client.html

**Comment by casualjim** [comment]
https://github.com/go-swagger/go-swagger/blob/master/examples/todo-list/client/todo_list_client.go#L51

**Comment by dlazerka** [comment]
Thank you, but those examples seem to be for some other swagger.

E.g. I downloaded client code at https://app.swaggerhub.com/apis/spire-ro/ro_api/
And there's no constructs like `api.Default.Operations.All(operations.AllParams{})` at all.
Manually investigating the the code I realized I need a completely different approach, something like:
```
cfg := api.NewConfiguration()
client := *api.NewAPIClient(cfg)
client.IonoApi.GetTecData(ctx, &opts)
```
There's nothing like that in examples by the links. Maybe there are some other examples? Wouldn't it be better to include those examples in README.md inside generated SDK package (like Java and Kotlin generators do)?
